server/src/core/distribution-report/distribution-report-service.ts

```diff
@@ -35,25 +39,110 @@ export class DistributionReports implements DistributionReportService {
     }
   }
 
-  private async sendDistributionReportMessages(reportDocs: DistributionReport[]): Promise<void> {
+  async sendMonthlyDistributionReportsToDonors(): Promise<void> {
+    try {
+      const lastMonthlyReportDate = await this.getLastMonthlyDistributionReportDate();
+      console.log('last monthly report date: ', lastMonthlyReportDate);
+      const report: MonthlyDistributionReport = await this.args.transactions.generateMonthlyDistributionReport(lastMonthlyReportDate);
+      await this.sendMonthlyDistributionReportMessages(report);
+      if (report.distributionReports.length) {
+        await this.collection.insertMany(report.distributionReports);
+      }
+    }
+    catch (e) {
+      console.error("Error occured: ", e.message);
+      rethrowIfAppError(e);
+      throw createDbOpFailedError(e.message);
+    }
+  }
+
+  private async sendDailyDistributionReportMessages(reportDocs: DistributionReport[]) {
     try {
       reportDocs.forEach(async (report: DistributionReport) => {
         const donor = await this.args.users.getById(report.donor);
         const beneficiaries = await this.getBeneficiaries(report.beneficiaries);
 
-        const amount = report.totalDistributedAmount < 2000 ? 2000 : report.totalDistributedAmount;
+        const amount = report.totalDistributedAmountFromDonor < 2000 ? 2000 : report.totalDistributedAmountFromDonor;
```

Comment:
Avoid using magic values. Create an appropriately-named constant variable to hold the value `2000`